### PR TITLE
feat(plugins): rewards plugin

### LINKED ISSUE #46: Generalized "GitHub Webhook + Contributor Role -> Rewards" No Config v1
Dynamically map the config property name to count the amount of matching webhook events that occur in the issue/pull timeline, and credit accordingly. In which case, this plugin seems generally useful for mapping any value to any event, in the context of an issue or pull! 

All this is intended to do is:
1. get the timeline of events from the issue and the linked pulls
2. assign a value of `1` to every event that is counted, per contributor.
3. return sum totals. 

In the next iteration, we should identify the user's "class" (specification author, assignee, collaborator, contributor)

In the final iteration, we should enable config. 

- All [webhook events](https://github.com/octokit/webhooks.js/blob/main/src/generated/webhook-names.ts) reference.

### PR DESCRIPTION
**Pull request by Amit0617**
Resolves [#46](https://github.com/ubiquity-os/plugins-wishlist/issues/46)

### CODE REVIEW

File: src/plugin.ts
Comment (Amit0617) [inline src/plugin.ts:1]: limiting scope here isn't meaningful when it is originally desired to be called on all kind of events

File: src/plugin.ts [hunk 1]
```diff
@@ -17,17 +18,8 @@ export async function runPlugin(context: Context) {
...
-  // increment the counter for each new event for every contributor in that issue or PR
-  const login = payload.sender?.login; // the user who triggered the event
-
-  if (login) {
-    if (contributors[login]) {
-      contributors[login]++;
-    } else {
-      contributors[login] = 1;
-    }
+  if (eventName === "issue_comment.created" && payload.comment.body.match(/rewards/i)) {
```
Comment (Amit0617) [inline src/plugin.ts:21]: limiting scope here isn't meaningful when it is originally desired to be called on all kind of events. Maintainer input needed.

File: src/types/plugin-inputs.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export const pluginSettingsSchema = T.Object(
     configurableResponse: T.String(),
     customStringsUrl: T.Optional(T.String()),
   },
-  { default: { configurableResponse: "Hello, world! Amit passed from here" } }
+  { default: { configurableResponse: "Hello, world! Amit passed from here, testing" } }
 );
```
Comment (Amit0617) [inline src/types/plugin-inputs.ts:27]: For self: don't forget to remove this too.

**Comment by 0x4007** [comment]
Against our template? No you do your own repo and do a pull against your own fork. Then you link it to the task (priced github issue)

**Pull request closed without merging**